api_schema/schema.go

```diff
@@ -64,22 +64,97 @@ type CreateLearnSessionRequest struct {
 	Name string `json:"name"`
 }
 
-// Also used as a model in specs_db.
 type LearnSession struct {
+	ID           akid.LearnSessionID `json:"id"`
+	Name         string              `json:"name"`
+	IdentityID   akid.IdentityID     `json:"identity_id"`
+	ServiceID    akid.ServiceID      `json:"service_id"`
+	CreationTime time.Time           `json:"creation_time"`
+
+	// Optional field whose presence indicates that the learn session is an
+	// extension to an existing API spec.
+	BaseAPISpecID *akid.APISpecID `json:"base_api_spec_id,omitempty"`
+
+	// HasMany relationship.
+	Tags []LearnSessionTag `json:"tags"`
+}
+
+func NewLearnSession(
+	ID akid.LearnSessionID,
+	Name string,
+	IdentityID akid.IdentityID,
+	ServiceID akid.ServiceID,
+	CreationTime time.Time,
+
+	BaseAPISpecID *akid.APISpecID,
+
+	Tags []LearnSessionTag,
+) *LearnSession {
+	return &LearnSession{
+		ID:           ID,
+		Name:         Name,
+		IdentityID:   IdentityID,
+		ServiceID:    ServiceID,
+		CreationTime: CreationTime,
+
+		BaseAPISpecID: BaseAPISpecID,
+
+		Tags: Tags,
+	}
+}
+
+// An extended version of LearnSession that includes extra information for
+// presentation in the web console when listing learn sessions.
+// XXX Should inherit from LearnSession, but Go. :(
+type ListedLearnSession struct {
 	tableName struct{} `pg:"learn_sessions"`
 
-	ID           akid.LearnSessionID `pg:"id,pk" json:"id"`
-	Name         string              `pg:"name" json:"name"`
-	IdentityID   akid.IdentityID     `pg:"identity_id" json:"identity_id"`
-	ServiceID    akid.ServiceID      `pg:"service_id" json:"service_id"`
-	CreationTime time.Time           `pg:"creation_time" json:"creation_time"`
```

Comment:
I'm not familiar with Go's ORM, but was the `pg:...` fragment intentionally removed here?

Reply:
Never mind, I see how this has been structured.